demo-linux/demo.cpp

```diff
@@ -120,19 +144,71 @@ void demo()
             exit(-1);
         }
     }
+
+    writeToEEPROM();
 }
 
 void usage()
 {
-    printf("Usage: $ demo-linux [--count n] [--log-level DEBUG|INFO|ERROR|NEVER]\n");
+    printf("Usage: $ demo-linux [--count n] [--log-level DEBUG|INFO|ERROR|NEVER] [--integration-time-ms n] [--raman-mode]\n");
     exit(1);
 }
 
 void parseArgs(int argc, char** argv)
 {
     for (int i = 1; i < argc; i++)
     {
-        if (!strcmp(argv[i], "--count"))
+        printf("current arg is %s\n", argv[i]);
+        if (!strcmp(argv[i], "--integration-time-ms")) 
+        {
+            if (i + 1 < argc)
+            {
+                int ms = atoi(argv[++i]);
+                wp_set_integration_time_ms(specIndex,ms);
+            }
+            else
+            {
+                printf("not enough args according to integration time\n");
+                usage();
+            }
+        }
+        else if (!strcmp(argv[i], "--raman-mode"))
+        {
+            init();
+            double dark_spectrum[pixels];
+            double corrected_spectrum[pixels];
+            double initial_spectrum[pixels];
+            int i;
+            string stringSpec;
+            if (WP_SUCCESS == wp_get_spectrum(specIndex, dark_spectrum, pixels))
+            {
+                printf("Obtained a dark spectra\n");
+            }
+            else {
+                printf("Did not obtain a dark spectra\n");
+            }
+            printf("Took dark spectrum.\n\n");
+            printf("About to enable laser\n");
+            usleep(3000000);
+            wp_set_laser_enable(specIndex, 1);
+            usleep(1000);
+            wp_get_spectrum(specIndex, initial_spectrum, pixels);
+            wp_set_laser_enable(specIndex, 0);
+            usleep(1000);
+            for (i = 0; i < pixels; i++)
+            {
+                corrected_spectrum[i] = -1*(initial_spectrum[i] - dark_spectrum[i]);
```

Comment:
What is the -1 for?  "initial - dark" is fine.

Reply:
This is because of what I was asking about. The dark appears as normal and the initial_spectrum appears as a dark when I plot them in excel. In order to just get the desired results, I multiplied by -1 since even though it's written as "initial-dark" it prints as if it's "dark-initial."